lib/modules/manager/npm/extract/locked-versions.ts

```diff
@@ -87,8 +88,41 @@ export async function getLockedVersions(
         )!;
       }
     } else if (pnpmShrinkwrap) {
-      logger.debug('TODO: implement pnpm-lock.yaml parsing of lockVersion');
+      logger.debug('Found pnpm lock-file');
       lockFiles.push(pnpmShrinkwrap);
+      if (!lockFileCache[pnpmShrinkwrap]) {
+        logger.trace('Retrieving/parsing ' + pnpmShrinkwrap);
+        lockFileCache[pnpmShrinkwrap] = await getPnpmShrinkwrap(pnpmShrinkwrap);
+      }
+      const { lockfileVersion } = lockFileCache[pnpmShrinkwrap];
+
+      // pnpm-version to pnpm lock-file version relation: https://github.com/pnpm/spec/tree/master/lockfile
+      if (lockfileVersion === 6) {
+        if (packageFile.extractedConstraints?.pnpm) {
+          //  if the latest 7.x release is compatible we need to increase the constraint
+          if (
+            semver.satisfies('7.32.0', packageFile.extractedConstraints.pnpm)
+          ) {
+            packageFile.extractedConstraints.pnpm += ' >=8';
+          }
+        } else {
+          packageFile.extractedConstraints!.pnpm = '>=8';
```

Comment:
```suggestion
          packageFile.extractedConstraints.pnpm = '>=8';
```
isn't it validated in outer if?

Reply:
No.